src/tools/list/list-organisation-details.tool.ts

```diff
@@ -0,0 +1,114 @@
+import { listXeroOrganisationDetails } from "../../handlers/list-xero-organisation-details.handler.js";
+import { getExternalLink } from "../../helpers/get-external-link.js";
+import { CreateXeroTool } from "../../helpers/create-xero-tool.js";
+
+const ListOrganisationDetailsTool = CreateXeroTool(
+  "list-organisation-details",
+  "Lists the organisation details from Xero. Use this tool to get information about the current Xero organisation.",
+  {},
+  async () => {
+    const response = await listXeroOrganisationDetails();
+    if (response.error !== null) {
+      return {
+        content: [
+          {
+            type: "text" as const,
+            text: `Error fetching organisation details: ${response.error || "Unknown error"}`,
+          },
+        ],
+      };
+    }
+
+    const organisation = response.result;
+
+    if (!organisation) {
+      return {
+        content: [
+          {
+            type: "text" as const,
+            text: "No organisation details found.",
+          },
+        ],
+      };
+    }
+
+    const resolvedExternalLinks = await Promise.all(
+      organisation.externalLinks?.map(async (link, index) => {
+        const deepLink = link.url
+          ? await getExternalLink(link.url)
+          : null;
+        return `${index + 1}. ${link.linkType}: ${deepLink || link.url}`;
+      }) || []
+    );
```

Comment:
Could this be simpler? It doesn't appear that any of the called functions are async, so all of this extra promise handling isn't necessary :)

```suggestion
    const resolvedExternalLinks = organisation.externalLinks?.map((link, index) => `${index + 1}. ${link.linkType}: ${link.url ? getExternalLink(link.url) : link.url}`) || []
```